src/components/ImageSearch.vue

```diff
@@ -2,24 +2,39 @@
     <div class="search-bar">
         <div v-if="loaded_search_data" class="container p-0">
             <div class="row">
-                <ImageSearchBar ref="searchBar" v-bind:availableKeywords="available_keywords" :currentQueryLabel.sync="current_keyword"
+                <ImageSearchBar ref="searchBar" v-bind:availableKeywords="available_keywords"
+                                :currentQueryLabel.sync="current_keyword"
                                 v-on:searchTriggered="searchTriggerd"></ImageSearchBar>
             </div>
             <div class="row">
                 <div class="col-12">
+                    <div v-on:click="reset_page" class="btn btn-sm btn-secondary" v-if="page_start>1">Jump to Start
+                    </div>
+                    <div v-on:click="final_page" class="btn btn-sm btn-secondary" v-if="(page_end+20)<results_end && bucket_lenght>0">Jump
+                        to End
+                    </div>
+                    <div v-if="canLoadMore" class="row mt-4 mb-4">
+                        <div class="col-12">
+                            <div v-on:click="load_less" class="btn btn-sm btn-secondary" v-if="page_number>1 && bucket_lenght>0"> Back</div>
+                            <div v-on:click="load_less2" class="btn btn-sm btn-secondary" v-if="page_number>2 && bucket_lenght>0">[{{page_number-2}} ]</div>
+                            <div v-on:click="load_less" class="btn btn-sm btn-secondary" v-if="page_number>1 && bucket_lenght>0">[{{page_number-1}} ] </div>
+                            <div class="btn btn-sm btn-secondary" v-if="bucket_lenght>0">Current Page: [ {{ page_number }} ]</div>
+                            <div v-on:click="load_more" class="btn btn-sm btn-secondary" v-if="(page_end+20)<results_end && bucket_lenght>0">[ {{page_number+1}} ] </div>
+                            <div v-on:click="load_more2" class="btn btn-sm btn-secondary" v-if="(page_end+40)<results_end && bucket_lenght>0">[ {{page_number+2}} ]</div>
+                            <div v-on:click="load_more" class="btn btn-sm btn-secondary" v-if="(page_end+20)<results_end && bucket_lenght>0">Next >> </div>
+                        </div>
+                    </div>
                     <div class="container p-0">
                         {{ message }}
                         <div id="result-box" class="row">
-                            <ImageResult v-for="(item, index) in flickrImageItems.slice(0, 100)" v-bind:key="index" v-bind:data="item" v-bind:label_list="available_keywords" v-bind:current_keyword="current_keyword" v-on:labelClicked="labelClicked"></ImageResult>
+                            <ImageResult v-for="item in flickrImageItems.slice(page_start,page_end)"
```

Comment:
you did remove the `(item, index)` which is needed according to the VueJS style guide